estimagic/optimization/nlopt_optimizers.py

```diff
@@ -141,6 +140,899 @@ def nlopt_neldermead(
     return out
 
 
+def nlopt_praxis(
+    criterion_and_derivative,
+    x,
+    *,
+    convergence_relative_params_tolerance=CONVERGENCE_RELATIVE_PARAMS_TOLERANCE,
+    convergence_absolute_params_tolerance=CONVERGENCE_ABSOLUTE_PARAMS_TOLERANCE,
+    convergence_relative_criterion_tolerance=CONVERGENCE_RELATIVE_CRITERION_TOLERANCE,
+    convergence_absolute_criterion_tolerance=CONVERGENCE_ABSOLUTE_CRITERION_TOLERANCE,
+    stopping_max_criterion_evaluations=STOPPING_MAX_CRITERION_EVALUATIONS,
+):
+    """Minimize a scalar function using principal-axis method.
+
+    This is a gradient-free local optimizer originally described in:
+    Richard Brent, Algorithms for Minimization without Derivatives
+    (Prentice-Hall, 1972). (Reprinted by Dover, 2002.). It assumes quadratic
+    form of the optimized function and repeatedly updates a set of conjugate
+    search directions.
+
+    The algorithm, is not invariant to scaling of the objective function and may
```

Comment:
Here you mention that this algorithm needs scaling but I think you still pass a copy of the default algorithm info where `needs_scaling=False`. I suggest you add the argument `algorithm_info` to `_minimize_nlopt` (with default `None`). 

In `_minimize_nlopt` you use the default algorithm info if None was passed, otherwise you use the one that was passed. 

Currently we don't use the info anywhere but in the long run we want to use it to automatically scale optimization problems that need scaling.

Reply:
Thanks.